### PR TITLE
Developing

### PR DESCRIPTION
**Pull request by carlos-buruel**

### CODE REVIEW

**Comment by carlos-buruel** [comment]
Bug Fixes
Fix the hyperlink for report error for queries. (b2dfeaa)
Fix theme color for items of list when theme color has changed. (9f7d0fa)
Fix data pivot when query is 2 columns. (8e45d83)
Limit row alert  for drill down query. (db98c66)
Set language parameter for the web services. (a977264)

Features
Dialog for hide and show columns. (86e7423)
Filter for tables. (0ae09af)
Message when all columns are invisible. (ebd0bb8)
SQL Dialog for dashboards. (c40c564)
Set multi language mode for project. (abdd790)

File: app/src/main/java/chata/can/chata_ai_api/main/PagerActivity.kt [hunk 1]
```diff
@@ -64,13 +67,13 @@ class PagerActivity: BaseActivity(R.layout.pager_activity)
...
-				else -> Pair(0, 0)
-			}
-			overridePendingTransition(pData.first, pData.second)
+//			val pData = when(floatingView.placement)
+//			{
+//				ConstantDrawer.LEFT_PLACEMENT -> Pair(R.anim.from_left_in, R.anim.from_right_out)
+//				ConstantDrawer.RIGHT_PLACEMENT -> Pair(R.anim.from_right_in, R.anim.from_left_out)
+//				else -> Pair(0, 0)
+//			}
+//			overridePendingTransition(pData.first, pData.second)
```
Comment (Ben-CHATA) [inline app/src/main/java/chata/can/chata_ai_api/main/PagerActivity.kt]: remove

File: app/src/main/java/chata/can/chata_ai_api/fragment/main/MainFragment.kt [hunk 2]
```diff
@@ -206,9 +210,12 @@ class MainFragment: BaseFragment(), View.OnClickListener, MainContract
 			MainRenderPresenter(context, this@MainFragment).run { initViews(llContainer) }
 			animationAlert = AnimationAlert(findViewById(R.id.rlAlert))
 
+			swQA = findViewById(R.id.swQA)
 			//swDemoData = findViewById(R.id.swDemoData)
```
Comment (Ben-CHATA) [inline app/src/main/java/chata/can/chata_ai_api/fragment/main/MainFragment.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/activity/dm/DMActivity.kt [hunk 3]
```diff
@@ -75,6 +76,8 @@ class DMActivity: AppCompatActivity(R.layout.activity_pager_options), View.OnCli
 	override fun onCreate(savedInstanceState: Bundle?)
 	{
 		super.onCreate(savedInstanceState)
+		setContentView(R.layout.activity_pager_options)
+//		window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/activity/dm/DMActivity.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/dialog/hideColumn/CustomAlertDialog.kt [hunk 4]
```diff
@@ -0,0 +1,293 @@
...
+
+	fun setColors()
+	{
+		context1.run {
+			ThemeColor.currentColor.run {
+				rlParent.setBackgroundColor(pDrawerBackgroundColor)
+				tvTitle.setTextColor(pDrawerTextColorPrimary)
+				tvColumnName.setTextColor(pDrawerTextColorPrimary)
+				tvVisibility.setTextColor(pDrawerTextColorPrimary)
+				//vBorder.setBackgroundColor(pDrawerBorderColor)
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/dialog/hideColumn/CustomAlertDialog.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt [hunk 5]
```diff
@@ -142,11 +167,15 @@ class WebViewHolder(
...
 				}
 			}
+			val aConfigs = aMutable.toMutableList()
 			queryBase?.let {
 				if (it.isGroupable &&
 					(R.id.ivColumn in aConfigs || R.id.ivStackedColumn in aConfigs))
 				{
-					Collections.swap(aConfigs, 0, if (R.id.ivPivot in aConfigs) 2 else 1)
+//					if (!it.hasHistorySaved)
+//					{
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt [hunk 6]
```diff
@@ -274,53 +282,68 @@ class WebViewHolder(
...
 	private fun processQueryBase(simpleQuery: QueryBase)
 	{
-		rvContentTop.visibility = if (simpleQuery.visibleTop) View.VISIBLE else View.INVISIBLE
-		if (simpleQuery.query.isNotEmpty())
+		if (simpleQuery.onlyHTML)
 		{
-			tvContentTop.visibility = visible
-			tvContentTop.text = simpleQuery.query
+			simpleQuery.onlyHTML = false
+			//simpleQuery.lastId = ""
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt [hunk 7]
```diff
@@ -274,53 +282,68 @@ class WebViewHolder(
...
+				}
 				val pData = when(iv.id)
 				{
 					R.id.ivTable ->
 					{
 						val idHide = lastId
 						lastId = "#idTableBasic"
-						Pair("'$idHide', '#idTableBasic', ''", queryBase.rowsTable)
+						Pair("'$idHide', '#idTableBasic', '#idTableBasic'", queryBase.rowsTable)
+//						Pair("TypeEnum.TABLE", queryBase.rowsTable)
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt [hunk 8]
```diff
@@ -418,6 +445,7 @@ class WebViewHolder(
 					requestLayout()
 					Handler(Looper.getMainLooper()).postDelayed({
 						loadUrl("javascript:hideTables(${pData.first});")
+//						loadUrl("javascript:updateData(${pData.first});")
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/fragment/dataMessenger/holder/webView/WebViewHolder.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/pojo/webView/Table.kt [hunk 9]
```diff
@@ -25,67 +25,65 @@ object Table
 				for (indexValue in aIndex)
 				{
 					val oColumn = aColumns[indexValue]
-					if (oColumn.isVisible)
-					{
-						val cell = aRow[indexValue]
+					//if (oColumn.isVisible) { }
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/pojo/webView/Table.kt]: Remove

File: chata_ai/src/main/java/chata/can/chata_ai/view/SwitchDM.kt [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+			addView(_view)
+			//endregion
+			//region
+			_tv = TextView(context).apply {
+				id = R.id.statusSwitch
+				layoutParams = LayoutParams(-2, -2).apply {
+					addRule(ALIGN_START, R.id.toggleSwitch)
+				}
+				margin(4f)
+				//setText("Q&A On")
```
Comment (Ben-CHATA) [inline chata_ai/src/main/java/chata/can/chata_ai/view/SwitchDM.kt]: Remove

**Comment by carlos-buruel** [comment]
1. who is responsible for the release?  
   Carlos Buruel Ortiz
2. Who reviewed the PR?  
   Vidhya
3. What is the Git Tag number?  
   1.1.1
4. What are the major changes for the release?
   • Dialog for hide and show columns. 
   • Filter for tables. 
   • Message when all columns are invisible.
   • SQL Dialog for dashboards. 
   • Set multi language mode for project.
5. How will the new features be verified?  
   They were reviewed by our testing.
6. What is the rollback procedure?  
   Revert to versions 1.1.1
7. Is this change breaking?  
   No

**Review by Ben-CHATA — APPROVED** [review]

**Pull request merged**